test/e2e/framework/framework.go

```diff
@@ -445,9 +449,51 @@ func (f *Framework) CreateNamespace(ctx context.Context, baseName string, labels
 	// fail to create serviceAccount in it.
 	f.AddNamespacesToDelete(ns)
 
+	if TestContext.DockerConfigFile != "" {
+		// dockerSecretName is the name of the secret containing Docker credentials
+		// needed for private image registry authentication. With the Secret created, the default service account (in the new namespace)
+		// is patched with the created secret and can then be referenced by all the pods spawned by E2E process, and repository authentication should be successful.
+		dockerSecretName := "registry-cred-" + string(uuid.NewUUID())
+		secret, err := f.createSecretFromDockerConfig(ctx, ns.Name, dockerSecretName)
+		if err != nil {
+			return ns, fmt.Errorf("failed to create secret [%s] from docker config file: %v", dockerSecretName, err)
+		}
+
+		serviceAccountClient := f.ClientSet.CoreV1().ServiceAccounts(ns.Name)
+		serviceAccountConfig := v1svc.ServiceAccount(serviceAccountName, ns.Name)
+		serviceAccountConfig.ImagePullSecrets = append(serviceAccountConfig.ImagePullSecrets, v1svc.LocalObjectReferenceApplyConfiguration{Name: &secret.Name})
+
+		svc, err := serviceAccountClient.Apply(ctx, serviceAccountConfig, metav1.ApplyOptions{FieldManager: "e2e-framework"})
+		if err != nil {
+			return ns, fmt.Errorf("failed to patch imagePullSecret [%s] to service account [%s]: %v", secret.Name, svc.Name, err)
+		}
+
+	}
+
 	return ns, err
 }
 
+// createSecretFromDockerConfig creates a secret using the private image registry credentials.
+// The credentials are provided by --docker-config-file flag.
+func (f *Framework) createSecretFromDockerConfig(ctx context.Context, namespace, dockerSecretName string) (*v1.Secret, error) {
+	contents, err := os.ReadFile(TestContext.DockerConfigFile)
+	if err != nil {
+		return nil, fmt.Errorf("error reading docker config file: %v", err)
+	}
+	auth := string(contents)
+
+	secret := &v1.Secret{
+		Data: map[string][]byte{v1.DockerConfigJsonKey: []byte(auth)},
+		Type: v1.SecretTypeDockerConfigJson,
+	}
+	secret.Name = dockerSecretName
+	Logf("create image pull secret %s", secret.Name)
+
+	_, err = f.ClientSet.CoreV1().Secrets(namespace).Create(ctx, secret, metav1.CreateOptions{})
+
+	return secret, err
```

Comment:
nit: I would expect the method to return the actual content of the created secret (= what was returned by Create), not what was sent to the apiserver.

This becomes required when using `GeneratedName` where `Name` is only set in the returned secret.